ad_corp.py

```diff
@@ -1,65 +1,26 @@
-#!/usr/bin/env python
+from ldap3 import Server, Connection, AUTO_BIND_NO_TLS, SUBTREE
 
-import json
-import ldap
-import os
-try:
-  from ldap.controls.pagedresults import SimplePagedResultsControl
-except ImportError:
-  from ldap.controls.libldap import SimplePagedResultsControl
 
-# 'ldaps://ad.example.com:636', make sure you always use ldaps
-ad_url             = os.environ.get('AD_URL')
-ad_basedn          = os.environ.get('AD_BASEDN')
-ad_binddn          = os.environ.get('AD_BINDDN')
-ad_bindpw          = os.environ.get('AD_BINDPW')
+class CompanyDirectory(object):
```

Comment:
This is much simpler!
Maybe add some [pydoc](https://www.python.org/dev/peps/pep-0257/)?